vmm/src/lib.rs

```diff
@@ -1511,6 +1532,117 @@ impl Vmm {
             .map_err(|e| VmmActionError::NetworkConfig(ErrorKind::User, e))
     }
 
+    fn update_net_device(
+        &mut self,
+        new_cfg: NetworkInterfaceUpdateConfig,
+    ) -> std::result::Result<VmmData, VmmActionError> {
+        if !self.is_instance_initialized() {
+            // VM not started yet, so we only need to update the device configs, not the actual
+            // live device.
+            let old_cfg = self
+                .network_interface_configs
+                .iter_mut()
+                .find(|&&mut ref c| c.iface_id == new_cfg.iface_id)
+                .ok_or(VmmActionError::NetworkConfig(
+                    ErrorKind::User,
+                    NetworkInterfaceError::DeviceIdNotFound,
+                ))?;
+
+            // Check if we need to update the RX rate limiter.
+            if let Some(new_rlim_cfg) = new_cfg.rx_rate_limiter {
+                if let Some(ref mut old_rlim) = old_cfg.rx_rate_limiter {
+                    // We already have an RX rate limiter set, so we'll update it.
+                    old_rlim.update_buckets(
+                        new_rlim_cfg.bandwidth.map(|b| b.into_token_bucket()),
```

Comment:
Can you replace `into_token_bucket` with the `Into` trait?

later edit - I saw the TODO below, and I agree with it :+1: